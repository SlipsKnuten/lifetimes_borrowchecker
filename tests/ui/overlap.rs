fn main() {
    let mut s = String::from("hi");
    let r = &s;
    let m = &mut s;
    println!("{r}");
    m.push('!');
}
