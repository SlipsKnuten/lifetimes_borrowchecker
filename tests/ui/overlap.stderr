error[E0502]: cannot borrow `s` as mutable because it is also borrowed as immutable
 --> tests/ui/overlap.rs:4:13
  |
3 |     let r = &s;
  |             -- immutable borrow occurs here
4 |     let m = &mut s;
  |             ^^^^^^ mutable borrow occurs here
5 |     println!("{r}");
  |                - immutable borrow later used here
